types/codespacer.go

```diff
@@ -0,0 +1,39 @@
+package types
+
+// Codespacer is a simple struct to track reserved codespaces
+type Codespacer struct {
+	next     CodespaceType
+	reserved map[CodespaceType]bool
+}
+
+// NewCodespacer generates a new Codespacer with the starting codespace
+func NewCodespacer() *Codespacer {
+	return &Codespacer{
+		next:     CodespaceType(1),
+		reserved: make(map[CodespaceType]bool),
+	}
+}
+
+// reserve reserves a specified codespace
+func (c *Codespacer) reserve(codespace CodespaceType) {
+	if codespace == c.next {
+		c.next++
+	} else if codespace < c.next || c.reserved[codespace] {
+		panic("Cannot reserve codespace - already taken!")
+	} else {
+		c.reserved[codespace] = true
+	}
+}
+
+// Register registers a provided codespace and returns it
+func (c *Codespacer) Register(codespace CodespaceType) CodespaceType {
+	c.reserve(codespace)
+	return codespace
+}
+
+// RegisterDefault registers and returns the next available codespace
+func (c *Codespacer) RegisterDefault() CodespaceType {
```

Comment:
RegisterNext

Reply:
And should pass in some codespace as argument, so it should start from some codespace.  That way if 1000 is taken, it goes to 1001 and so on.

Reply:
Updated, now registers next available and panics if maximum (2^16 - 1) reached